src/AdoNetCore.AseClient/Internal/SchemaTableBuilder.cs

```diff
@@ -163,28 +168,41 @@ private void TryLoadKeyInfo(DataTable table, string baseTableNameValue, string b
 
                 try
                 {
+                    var columnMetadata = new List<ColumnMetadata>();
                     using (var keyInfoDataReader = command.ExecuteReader())
                     {
                         while (keyInfoDataReader.Read())
                         {
-                            var keyColumnName = keyInfoDataReader["COLUMN_NAME"].ToString();
-                            var keySchemaName = keyInfoDataReader["TABLE_SCHEMA"].ToString();
-                            var keyCatalogName = keyInfoDataReader["TABLE_CATALOG"].ToString();
+                            columnMetadata.Add(new ColumnMetadata
+                            {
+                                Name = keyInfoDataReader["COLUMN_NAME"].ToString(),
+                                Schema = keyInfoDataReader["TABLE_SCHEMA"].ToString(),
+                                Catalog = keyInfoDataReader["TABLE_CATALOG"].ToString(),
+                                IsKey = (bool) keyInfoDataReader["PRIMARY_KEY"],
+                                IsUnique = (bool) keyInfoDataReader["UNIQUE"]
+                            });
+                        }
+                    }
 
-                            foreach (DataRow row in table.Rows)
+                    // Check for duplicate name and if we found any then give it a miss
+                    if (!columnMetadata.Any() || columnMetadata.GroupBy(c => c.Name)
```

Comment:
Related to avoiding Linq: You could probably just use a `Dictionary<string, ColumnMetadata>()`.
Before inserting do a `.ContainsKey()` and return early.
Later on you'd just loop through `.Values`